src/collection.js

```diff
@@ -1422,6 +1439,19 @@ export class CollectionTransaction {
     return { data: existing, permissions: {} };
   }
 
+  /**
+   * Soft Delete all records from the local database.
+   * @param  {Object} allRecords       All the records in the collection.
+   * @return {Object}
+   */
+  deleteAll(allRecords) {
+    allRecords.forEach(record => {
+      delete record.id;
+    });
+    this._queueEvent("deleteAll", { data: [] });
```

Comment:
I believe it would be more useful for developers to have information about the deleted records instead of an empty list

Maybe, instead of adding a new `deleteAll` event, we could emit an individual `delete` event for each delete record

Reply:
`delete` events will be emitted since I'm using the delete method, which emits delete. I thought you might want to know that `deleteAll` is triggered for future reference!

Reply:
yes OK

Reply:
But we could add a test then (that asserts we receive events for individual records)

Reply:
Good one, I'll do that then